### PR TITLE
[feat] Add hidden states extraction for TorchSpec training

### PR DESCRIPTION
**Pull request by sijyang**
## Summary

This PR adds TorchSpec integration support to ATOM and includes two related improvements to the inference engine.

### 1. TorchSpec Hidden States Extraction

Extend `RLHFModelRunner` with opt-in hidden states extraction (`configure_hidden_states()`), supporting two capture strategies:

- **Native model forward capture** via `capture_hidden_state_layers` kwarg (implemented for Qwen3).
- **Hook-based capture** as automatic fallback for models that do not expose the kwarg — registers `forward_hook` on decoder layers to intercept post-layer residual streams.

Captured hidden states (auxiliary layers + last hidden) are written to **Mooncake KV store** per-request during prefill, keyed by `external_request_id`.

**New APIs on `AsyncLLMEngine`:**

- `configure_hidden_states(aux_layer_ids, mooncake_config)` — enable extraction and connect to Mooncake.
- `generate_hidden_states(input_ids_list, data_ids)` — run prefill-only forward and write hidden states to Mooncake.

**Plumbing:**

- `configure_hidden_states` utility command through `EngineUtilityHandler` for cross-process configuration.
- `ScheduledBatch.external_request_ids` populated from sequence metadata for Mooncake key mapping.

### 2. MTP Statistics API

Add a structured HTTP endpoint to expose speculative decoding acceptance statistics at runtime.

- `GET /debug/mtp_stats` — returns aggregated MTP statistics across all DP ranks.
- `LLMEngine.get_mtp_statistics()` broadcasts to all engine cores and merges results.
- New `get_mtp_statistics` command in `EngineCore` returning a structured dict (vs. existing `get_mtp_stats` which only prints to console).

#### Usage

With an ATOM server running speculative decoding (MTP), query acceptance statistics via HTTP:

```bash
curl http://localhost:8000/debug/mtp_stats
```

Example response:

```json
{
  "enabled": true,
  "total_draft_tokens": 102400,
  "total_accepted_tokens": 71680,
  "acceptance_rate": 0.70,
  "average_tokens_per_forward": 1.70,
  "distribution": {
    "0": 3072,
    "1": 10240,
    "2": 38400
  },
  "distribution_percent": {
    "0": 0.06,
    "1": 0.20,
    "2": 0.74
  }
}
```

**Field reference:**

| Field | Description |
|-------|-------------|
| `enabled` | Whether MTP is active |
| `total_draft_tokens` | Cumulative draft tokens proposed |
| `total_accepted_tokens` | Cumulative draft tokens accepted by the verifier |
| `acceptance_rate` | `accepted / draft` |
| `average_tokens_per_forward` | Mean tokens produced per forward step (`1 + accepted / steps`) |
| `distribution` | Per-step accepted token count histogram (key = number accepted, value = occurrences) |
| `distribution_percent` | Same histogram in percentage form |

> **Note:** In multi-DP-rank deployments, the response is aggregated across all ranks. When MTP is not enabled, the endpoint returns `{"enabled": false}`.

### 3. Logprobs Bugfix

Remove temperature scaling from logprobs computation. Logprobs should reflect the model's raw log-probability of the sampled token, not the temperature-adjusted distribution used for sampling. Previously, `log_softmax` was applied on temperature-scaled logits, producing incorrect logprob values (especially with high temperatures).

## Changed Files

| File | Change |
|------|--------|
| `atom/models/qwen3.py` | Add `capture_hidden_state_layers` support to `Qwen3Model.forward()` |
| `atom/rollout/model_runner_ext.py` | Hidden states capture logic in `RLHFModelRunner` (hook registration, `run_model` override, Mooncake store write) |
| `atom/rollout/async_engine.py` | `configure_hidden_states()` and `generate_hidden_states()` on `AsyncLLMEngine` |
| `atom/rollout/engine_utility.py` | `configure_hidden_states` utility command handler |
| `atom/model_engine/scheduler.py` | Populate `external_request_ids` in `ScheduledBatch` |
| `atom/entrypoints/openai/api_server.py` | `GET /debug/mtp_stats` HTTP endpoint |
| `atom/model_engine/llm_engine.py` | `get_mtp_statistics()` with DP rank aggregation |
| `atom/model_engine/engine_core.py` | `get_mtp_statistics` command handler returning structured result |
| `atom/model_engine/model_runner.py` | Remove temperature scaling from logprobs computation |

### CODE REVIEW

**Comment by valarLip** [comment]
@copilot resolve the merge conflicts in this pull request

**Review by valarLip — APPROVED** [review]
